cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ApplicationSpecificationAdapter.java

```diff
@@ -165,6 +165,11 @@ private ApplicationSpecificationAdapter(SchemaGenerator schemaGenerator, Gson gs
         SortedMap.class.isAssignableFrom(rawType)) {
         return null;
       }
+
+      if (!ParameterizedType.class.isAssignableFrom(type.getType().getClass())) {
```

Comment:
When I want to update the run record with the WorkflowNodeState, the corresponding code is 
```java
    Map<String, String> propertiesToUpdate = new HashMap<>();
    propertiesToUpdate.putAll(runRecordMeta.getProperties());
    propertiesToUpdate.put(WORKFLOW_NODE_STATE_KEY, GSON.toJson(nodeStates));
```
Conversion of `nodeStates` to json using GSON instance fails. This happens because when the `ApplicationSpecificationAdapter.create` is called type parameter passed to it is of type `java.util.HashMap` casting of which to `ParameterizedType` fails. 

Another approach to fix it is use `new Gson()` instead of static `GSON`.